google/cloud/storage/oauth2/service_account_credentials.h

```diff
@@ -37,6 +38,10 @@ struct ServiceAccountCredentialsInfo {
   std::string private_key_id;
   std::string private_key;
   std::string token_uri;
+  // Optional; if empty, a default set of scopes will be used.
+  std::set<std::string> scopes;
+  // See https://developers.google.com/identity/protocols/OAuth2ServiceAccount.
+  std::string subject;
```

Comment:
Perhaps std::optional here too? It'll make the code itself almost as clear as the comments.

Reply:
Done, good call.